src/cdomains/intDomain.ml

```diff
@@ -2307,16 +2307,36 @@ struct
     if M.tracing then  M.trace "congruence" "shift_left : %a %a becomes %a \n" pretty x pretty y pretty res;
     res
 
+  (* Handle unsigned overflows. 
+     From n === k mod (2^a * b), we conclude n === k mod 2^a, for a <= 2^{bitwidth}.
+     The congruence modulo b may not persist on an overflow. *)
+  let handle_overflow ik (c, m) =
+    let max = (snd (Size.range ik)) +: Ints_t.one in
+    (* Find largest m'=2^k (for some k) such that m is divisible by m' *)
+    let tz = Ints_t.trailing_zeros m in
+    let m' = Ints_t.shift_left (Ints_t.of_int 1) tz in
```

Comment:
We need a special case here since an intdomain property test is failing. This is because of a special case:

>  If `x` is zero, `trailing_zeros x` returns `max_int`.

Reply:
Thanks for the hint! `m=0` is actually only passed to `handle_overflow` when one of the operands of `mul` is `0` and `no_ov = false`. This could e.g. happen when mutual refinement between domains is not active, so it's good to handle this.